### PR TITLE
Add biosecurity admin features #191

### PR DESCRIPTION
**Pull request by yasima-csiro**

### CODE REVIEW

**Comment by brucehyslop** [comment]
It looks like the changes allows users with the `ROLE_BIOSECURITY_ADMIN` role to list all users. 

Should biosecurity admin uses be able to view all user details? If not we could put the "New User" on the admin page or
prevent the listing (empty user list) on the find user page.

**Comment by yasima-csiro** [comment]
Thanks Bruce. I was thinking that biosecurity admin may want to search users with email first before creating users. 

@sbearcsiro @sat01a WDYT?

**Comment by yasima-csiro** [comment]
According to the discussion had with Sathish, decided to remove list/search user permission for ROLE_BIOSECURITY_ADMIN role.

**Comment by sat01a** [comment]
Thanks Yasima

**Comment by yasima-csiro** [comment]
@brucehyslop I addressed the review comments. Would you be able to check? Thanks.

**Review by brucehyslop — APPROVED** [review]
looks good

File: src/main/groovy/au/org/ala/auth/PreAuthorise.groovy [hunk 1]
```diff
@@ -30,7 +30,11 @@ import java.lang.annotation.Target
 @Retention(RetentionPolicy.RUNTIME)
 @Documented
 public @interface PreAuthorise {
-    String requiredRole() default "ROLE_ADMIN"
+    /***
+     *  A list of roles that is allowed to access the method. The user must have at least one role to access the method.
+     * @return
+     */
+    String[] requiredRoles() default ["ROLE_ADMIN"]
```
Comment (sbearcsiro) [inline src/main/groovy/au/org/ala/auth/PreAuthorise.groovy:37]: Nit: requiredRoles implies (to me) that all roles are required.  Suggest a change of name to something like `allowedRoles`

**Review by sbearcsiro — COMMENTED** [review]
This looks good but I wonder if we shouldn't name the role after the capability (ie ROLE_USER_CREATOR) rather than the client application, so that we can give this ability to other apps without giving the user biosecurity admin or making a code change?

**Comment by sat01a** [comment]
> This looks good but I wonder if we shouldn't name the role after the capability (ie ROLE_USER_CREATOR) rather than the client application, so that we can give this ability to other apps without giving the user biosecurity admin or making a code change?

I like this suggestion, @qifeng-bai and @yasima-csiro ?

**Comment by yasima-csiro** [comment]
I am ok with that.

**Comment by qifeng-bai** [comment]
@yasima-csiro 
ROLE_BIOSECURITY_ADMIN  is only Biosecurity admins to subscribe Biosecurity query and manage subscribers of Biosecurity ONLY

ROLE_USER_CREATOR  is more general. To me, the role is for user creation?

**Comment by yasima-csiro** [comment]
Thanks Bai. Yes, ROLE_USER_CREATOR role is permitted only to create users. I am happy to use this role in userdetails to make it reusable and not to tight it with alerts app. 

In that case, Biosecurity admins should have both roles - ROLE_BIOSECURITY_ADMIN and ROLE_USER_CREATOR to manage biosecurity alerts and create users in userdetails.

WDYT @sbearcsiro @sat01a @qifeng-bai @brucehyslop ?

**Comment by sat01a** [comment]
> Thanks Bai. Yes, ROLE_USER_CREATOR role is permitted only to create users. I am happy to use this role in userdetails to make it reusable and not to tight it with alerts app.
> 
> In that case, Biosecurity admins should have both roles - ROLE_BIOSECURITY_ADMIN and ROLE_USER_CREATOR to manage biosecurity alerts and create users in userdetails.
> 
> WDYT @sbearcsiro @sat01a @qifeng-bai @brucehyslop ?

Yes agreed Yasima. @kylie-m  will make a note of these new roles.

**Comment by kylie-m** [comment]
Draft documentation now on Confluence pages:
- [Alerts](https://confluence.csiro.au/display/ALASD/Alerts)
- [Biosecurity Alerts admin page functionality](https://confluence.csiro.au/pages/viewpage.action?pageId=1913483912)
- [Biosecurity Alerts workflow](https://confluence.csiro.au/display/ALASD/ALA+Biosecurity+Alerts+Workflow)

**Review by sbearcsiro — APPROVED** [review]
LGTM

**Pull request merged**